poll/app/ui-components/cards/poll-list.tsx

```diff
@@ -0,0 +1,145 @@
+"use client";
+
+import { EditIcon } from "@/public/svgs/edit";
+import { DeleteIcon } from "@/public/svgs/delete";
+import { ShareIcon } from "@/public/svgs/share";
+// import { toast } from "react-toastify";
+import Link from "next/link";
+
+interface pollData {
+    poll_title: string;
+    poll_description: string;
+    poll_question: string;
+    poll_type: string;
+    poll_options: pollOption[];
+}
+
+interface pollOption {
+    id: string;
+    label: string;
+    value: string;
+}
+
+interface PollListProps {
+    allPolls: pollData[];
+    setAllPolls: (polls: pollData[]) => void;
+    handleEditCreatedPoll: (index: number) => void;
+}
+
+export const PollList = ({
+    allPolls,
+    setAllPolls,
+    handleEditCreatedPoll,
+}: PollListProps) => {
+    const handleDeletePoll = (index: number) => {
+        const isConfirmed = window.confirm(
+            "Are you sure you want to delete this poll?"
+        );
+        if (isConfirmed) {
+            const updatedPollList = allPolls.filter((_, idx) => idx !== index);
+            setAllPolls(updatedPollList);
+
+            // update localStorage with the new list of polls
+            localStorage.setItem(
+                "createpolldata",
+                JSON.stringify(updatedPollList)
+            );
+        }
+    };
+
+    const handleEditClick = (index: number) => {
+        handleEditCreatedPoll(index);
+    };
+
+    return (
+        <div>
+            <h2 className="text-dark-gray mb-4 font-bold">Polls</h2>
+            {allPolls.map((poll, index) => (
+                <details
+                    key={index}
+                    className="mb-6 rounded-lg border border-[#e1e1e1] p-4"
+                >
+                    <summary className="cursor-pointer font-semibold">
+                        {poll.poll_question}
```

Comment:
The title would be better here as the question may be quite long and the title should be a quite short name, that in most cases is shorter than the question
```suggestion
                        {poll.poll_title}
```